internal/core/src/storage/GrowingMmapChunkManager.h

```diff
@@ -0,0 +1,176 @@
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include <cstdint>
+#include <cstring>
+#include <vector>
+#include <queue>
+#include <atomic>
+#include <unordered_map>
+#include <shared_mutex>
+#include <memory>
+#include <shared_mutex>
+#include "common/EasyAssert.h"
+namespace milvus::storage {
+/**
+ * @brief keeping all information of tmp files
+ */
+struct MappState {
+ public:
+    MappState(const uint64_t disk_limit,
+              const uint64_t fix_file_size,
+              const std::string file_prefix)
+        : max_disk_limit(disk_limit),
+          mmap_file_prefix(file_prefix),
+          fix_mmap_file_size(fix_file_size) {
+        mmapp_file_counter.store(0);
+    }
+    std::string
+    GetMmapFilePath() {
+        auto file_id = mmapp_file_counter.fetch_add(1);
+        return mmap_file_prefix + "/" + std::to_string(file_id);
+    }
+    uint64_t
+    GetDiskLimit() {
+        return max_disk_limit;
+    }
+    uint64_t
+    GetFixFileSize() {
+        return fix_mmap_file_size;
+    }
+    std::string
+    GetFilePrefix() {
+        return mmap_file_prefix;
+    }
+
+ private:
+    uint64_t max_disk_limit;
+    std::string mmap_file_prefix;
+    std::atomic<uint64_t> mmapp_file_counter;
+    uint64_t fix_mmap_file_size;
+};
+using MappStatePtr = std::shared_ptr<MappState>;
+
+/**
+ * @brief MappEntry handle all memory mapping in one tmp file
+ */
+struct MappEntry {
+ public:
+    enum class EntryType {
+        NORMAL = 0,
+        LARGE = 1,
+    };
+    // create a new mmap file and init some objs
+    MappEntry(const std::string& file_name,
+              const uint64_t file_size,
+              EntryType type = EntryType::NORMAL);
+    ~MappEntry();
+    void
+    Clear() {
+        offset_.store(0);
+    }
+    void*
+    Allocate(const uint64_t size, ErrorCode& error_code);
+    void
+    Reset() {
+        return offset_.store(0);
+    }
+    EntryType
+    GetType() {
+        return entry_type_;
+    }
+    uint64_t
+    GetCapacity() {
+        return file_size_;
+    }
+
+ private:
+    const std::string file_name_;
+    const uint64_t file_size_;
+    char* addr_ = nullptr;
+    std::atomic<uint64_t> offset_ = 0;
+    const EntryType entry_type_;
+};
+using MappEntryPtr = std::shared_ptr<MappEntry>;
+
+/*
+* @brief keeping free tmp files in a queue 
+*/
+class FixedSizeMmapEntryQueue {
```

Comment:
Ambiguous name and redundant layer. We can:
1. Merge it with `GrowingMmapChunkManager`
2. Make a cache module for `GrowingMmapChunkManager` instead a layer between it and the resource.
BTW, `GrowingMmapChunkManager` can be rename to `MmapChunkManager`